source/slang/slang-emit-glsl.cpp

```diff
@@ -1470,6 +1470,19 @@ void GLSLSourceEmitter::emitEntryPointAttributesImpl(
     default:
         break;
     }
+
+    if (irFunc->findDecoration<IRQuadDerivativesDecoration>())
```

Comment:
This is starting to get too many if (findDecoration) blocks, note that each one of them is a linear search, we may just use one single for loop to interate and test the op code of all decorations.